internal/mage/engine.go

```diff
@@ -223,18 +234,16 @@ func (t Engine) test(ctx context.Context, race bool) error {
 		tests = tests.WithEnvVariable("_EXPERIMENTAL_DAGGER_CACHE_CONFIG", cacheEnv)
 	}
 
-	output, err := tests.
+	_, err = tests.
 		WithMountedFile(cliBinPath, util.DaggerBinary(c)).
 		WithEnvVariable("_EXPERIMENTAL_DAGGER_CLI_BIN", cliBinPath).
 		WithEnvVariable("_EXPERIMENTAL_DAGGER_RUNNER_HOST", endpoint).
 		WithMountedDirectory("/root/.docker", util.HostDockerDir(c)).
+		WithExec([]string{"go", "install", "gotest.tools/gotestsum@v1.10.0"}).
```

Comment:
Debated creating a `tools.go` and tracking this in our `go.mod`, but this seems OK to me too. I don't think we really care to keep its dependencies in the same set as ours, and it's at least pinned, and will work across all platforms (as opposed to grabbing a binary from their GitHub release).